### PR TITLE
scripts: refactored `daemon-stack.sh`

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

The file `target/scripts/startup/daemons-stack.sh` is refactored with this PR. It was cluttered with Fetchmail setup, which was relocated. A common daemon startup routine was introduced with proper logging. The log will now show startup information on log level `debug` or `trace`, and output from Supervisor is only shown on error. The currently present

``` TXT
<DAEMON>: started
```

is therefore not shown anymore during startup. On the right log level, the information can be obtained:

``` TXT
[  DEBUG  ]  Starting <DAEMON>
```

Functionality-wise, this PR changes nothing. This is just a refactoring.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: target/scripts/startup/daemons-stack.sh [hunk 1]
```diff
@@ -1,160 +1,86 @@
...
+function _start_daemon_dovecot
 {
-  _log 'debug' 'Starting dovecot services'
-
   if [[ ${ENABLE_POP3} -eq 1 ]]
   then
-    _log 'debug' 'Starting pop3 services'
-    mv /etc/dovecot/protocols.d/pop3d.protocol.disab \
-      /etc/dovecot/protocols.d/pop3d.protocol
+    _log 'debug' 'Starting POP3 services'
```
Comment (polarathene) [inline target/scripts/startup/daemons-stack.sh:64]: Technically it's enabling it through config for Dovecot to manage?

```suggestion
    _log 'debug' 'Enabling POP3 services'
```

File: target/scripts/startup/daemons-stack.sh [hunk 2]
```diff
@@ -1,160 +1,86 @@
...
-command=/usr/bin/fetchmail -f ${RC} -v --nodetach --daemon %(ENV_FETCHMAIL_POLL)s -i /var/lib/fetchmail/.fetchmail-UIDL-cache --pidfile /var/run/fetchmail/%(program_name)s.pid
-EOF
-      chmod 700 "${RC}"
-      chown fetchmail:root "${RC}"
-    done
-
-    supervisorctl reread
-    supervisorctl update
-
     COUNTER=0
```
Comment (polarathene) [inline target/scripts/startup/daemons-stack.sh:77]:
```suggestion
    local COUNTER=0
```

**Review by polarathene — DISMISSED** [review]
These diffs were not pleasant to review (at least via the Github UI). I went through individual commits, but these were not tailored to improve the diff experience :sweat_smile: 

LGTM though, nice improvements :+1:

**Comment by georglauterbach** [comment]
> These diffs were not pleasant to review (at least via the Github UI). I went through individual commits, but these were not tailored to improve the diff experience sweat_smile

Yes, the diff is very ugly...
 
> LGTM though, nice improvements +1

Thanks!

**Comment by georglauterbach** [comment]
> Test failure:
> 
> ```
> not ok 300 checking setup.sh: setup.sh debug fetchmail in [315](https://github.com/docker-mailserver/docker-mailserver/runs/5625535875?check_suite_focus=true#step:8:315)ms
> # (from function `assert_output' in file test/test_helper/bats-assert/src/assert.bash, line 231,
> #  in test file test/tests.bats, line 1128)
> #   `assert_output --partial "fetchmail: normal termination, status 11"' failed
> # 
> # -- output does not contain substring --
> # substring (1 lines):
> #   fetchmail: normal termination, status 11
> # output (2 lines):
> #   /usr/local/bin/debug-fetchmail: line 3: /usr/local/bin/setup-fetchmail: No such file or directory
> #   fetchmail: no mailservers have been specified.
> # --
> ```
> 
> Since you migrated the fetchmail scripts to `setup-stack.sh`, you now have this failure trying to run `_setup_fetchmail()` that you'll need to resolve:
> 
> https://github.com/docker-mailserver/docker-mailserver/blob/00dffb21f0d3de98a1fbecc4108154e2a97acaa6/target/bin/debug-fetchmail#L3

I just sourced `setup-stack.sh`. The commit message again explains why this is fine (sourcing the stack does not execute code until one calls the functions themselves).

**Review by polarathene — DISMISSED** [review]
> I just sourced `setup-stack.sh`.

Still feels icky :sweat_smile: but I'm fine with it.

**Comment by georglauterbach** [comment]
@casperklein what do you think about this PR? :)

File: target/bin/debug-fetchmail [hunk 3]
```diff
@@ -1,6 +1,9 @@
 #! /bin/bash
 
-/usr/local/bin/setup-fetchmail
+# shellcheck source=../scripts/startup/setup-stack.sh
+source /usr/local/bin/setup-stack.sh
```
Comment (casperklein) [inline target/bin/debug-fetchmail:4]: `_setup_fetchmail` uses `_log` which isn't available, when just sourcing `setup-stack.sh`.

```suggestion
# shellcheck source=../scripts/helpers/log.sh
source /usr/local/bin/helpers/log.sh
# shellcheck source=../scripts/startup/setup-stack.sh
source /usr/local/bin/setup-stack.sh
```

**Review by casperklein — CHANGES_REQUESTED** [review]

File: target/bin/debug-fetchmail (hunk 3 above)
Comment (georglauterbach) [reply target/bin/debug-fetchmail:6]: Nice catch! I resolved the conflicts with `master` and then included these changes as well. Done in ba8b6261a85428275435cd279ca18124d84b42d4

**Review by polarathene — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**